cirq-core/cirq/protocols/decompose_protocol.py

```diff
@@ -364,6 +367,11 @@ def decompose_once(
         TypeError: `val` didn't have a `_decompose_` method (or that method returned
             `NotImplemented` or `None`) and `default` wasn't set.
     """
+    if context is None:
+        context = DecompositionContext(
+            ops.SimpleQubitManager(prefix=f'_decompose_protocol_{time.time_ns()}')
```

Comment:
Can we use some global counter instead of time_ns here?
It would be nice to have naming reproducibility in repeated executions.
Debugging and making sense of allocated qubit names should get easier.

Reply:
added using `itertools.count()`